### ISSUE TITLE
Google keeps saying we're running out of Google Maps API credits

### ISSUE THREAD

**Issue by henare** [openaustralia/planningalerts#1122]
Google Maps is an integral part of PlanningAlerts. We display Maps and StreetView on application pages, include static maps and StreetView images in email alerts, and use the geocoder to turn messy addresses from councils into latitude and longitude.

On 2014-10-02 Google emailed to say we'd reached 100% of our quota. As far as I know nothing happened and everything kept working.

On 2015-11-05 we were again told we'd reached 100% of quota. On the 24th this said 110% of quota. Again I don't think anything happened and all was well.

On 2016-10-11 we were advised we'd used 75% of our quota. I replied and asked for our quota to be increased. We were told we had to go through the whole process of applying again, which we did, and after about a month we were told our quota had been upgraded from 1,000,000 credits to 2,000,000.

On 2017-01-19 we got an email that said, "You have used 25% of your Maps API quota in less than 60 days". This then increased to 50% used on 2017-03-09. The email also said that our license start date was Jan 18, 2017 which would have meant we were only 2 months into our license and had already used 50% of our quota. Not good!

I've subsequently contacted Google support and they've clarified that the Usage Start Date is actually November 3rd, 2016 because that's when we hit 100% of our previous license. This should mean we've used 50% of our credits in 4 months out of a 12 month licence but I've emailed Google to clarify.

## What's using our credits?

When I first started looking into this I assumed that geocoding would have been using all our credits. That's definitely not the case - you get a [shared 100,000 requests per day quota](https://developers.google.com/maps/premium/usage-limits#web-service-apis) of these kinds of requests and we only do around 6,000.

It appears that Map loads is the source. We currently have about 6,500 pageviews per day below `/applications`. Loading a map uses 1 credit so that's **at least 2,372,500 credits per year - more than our quota**.

Since we also have StreetView on those pages does that mean we're actually using 2 credits on that page?

**Comment by henare** [comment]
>It appears that Map loads is the source. We currently have about 6,500 pageviews per day below /applications. Loading a map uses 1 credit so that's at least 2,372,500 credits per year - more than our quota.

The Google Maps free plan - for sites that free like PlanningAlerts - [allows 25,000 map loads per day](https://developers.google.com/maps/pricing-and-plans/#details). Are we just doing it wrong?

**Comment by henare** [comment]
We've implemented a pretty big change. We're now using the free plan for our map loads like I think we're supposed to be.

At our current API usage rate of an average of 8,559 credits/day since our license started we would be due to exhaust our quota on the 27th of June. We should check in again and see how this change has affected our usage. Success is if we've reined in usage enough to last until November the 3rd (which is when our licence gets renewed, I think).

**Comment by henare** [comment]
>I've subsequently contacted Google support and they've clarified that the Usage Start Date is actually November 3rd, 2016 because that's when we hit 100% of our previous license. This should mean we've used 50% of our credits in 4 months out of a 12 month licence but I've emailed Google to clarify.

They've responded and I think they're saying the start date is actually January 18th:

>Pulling up your purchase in my tools, I see a Support End Date of Jan 18th, 2018 for Premium License asset gme-openaustraliafoundation & PROJ-[redacted] (EXT).
>
>You should receive the full term of your license, as touched on in your welcome letter. The subject line for this e-mail was "Welcome to the Google Maps APIs Premium Plan! - OPENAUSTRALIA FOUNDATION LIMITED" and it was sent out on Jan 18th, 2017.

So what I'm hearing is that our credits started to get used on November 3 2016 but our 1 year plan started on January 18 2017 and that will expire on January 18 2018. This means we've got less than 1,000,000 credits available to us for the next 10 months :sweat:

**Comment by kat** [comment]
So @henare it still  looks like we use our quota by 27th of June? What are our options?

**Comment by henare** [comment]
@kat we need to see the results of the change I made. I'll update the issue as I progress.

**Comment by henare** [comment]
The changes had a massive impact - that was definitely the problem:

![selection_001](https://cloud.githubusercontent.com/assets/48945/24086534/1280ebda-0d65-11e7-91a3-fbc2402dbacc.png)

However I noticed that autocomplete and geolocating weren't working - I needed to switch on the  Google Places API Web Service in our API console. This very quickly blew out the 1,000 free requests per day so I also had to enable billing. Apparently this is just for verification purposes and we now get 150,000 free requests per day.

This means we're using the free account, within limits, for everything except geocoding. Because our application is freely available I think we're doing this within the rules. We do about 6,000 geocodes a day and that's beyond the 2,500 you get for free so it seems reasonable to use our Premium account for that - you get 150,000 on a premium account without using any credits so that should be OK.

I'll keep this issue open for another week to keep an eye on things but I think we're now in the clear.
